### PR TITLE
Add third-party CA enrollment integration tests

### PR DESCRIPTION
**Pull request by CLBRITTON2**
Adds integration tests for enrolling with client certificates signed by a registered 3rd-party CA: 
- Auto CA enrollment (no pre-created identity)
- OTT CA enrollment (a one-time token presented with a client certificate signed by the CA)
- Rejection of a certificate signed by an unregistered CA.

### CODE REVIEW

File: tests/integration/testutil/overlay.go [hunk 1]
```diff
@@ -555,26 +567,135 @@ func (o *Overlay) CreateIdentityWithExternalId(t *testing.T, name, externalID, a
...
+// pki CLI lays out each CA at <root>/<ca-name>.
+func (o *Overlay) pkiRoot() string {
+	return filepath.Join(o.Home, "third-party-pki")
+}
+
+// CreateLocalPkiCA creates a root CA on disk without registering it on the
+// controller.
+func (o *Overlay) CreateLocalPkiCA(t *testing.T, name string) {
+	// ziti pki refuses to overwrite PKI left by a previous run
+	require.NoError(t, os.RemoveAll(filepath.Join(o.pkiRoot(), name)), "clear stale pki for %s", name)
```
Comment (dovholuknf) [inline tests/integration/testutil/overlay.go]: don't love that it removeall's here. seems like the 'create' func shouldn't do destructive operations without a boolean at least passed to do so

File: tests/integration/testutil/overlay.go [hunk 2]
```diff
@@ -555,26 +567,135 @@ func (o *Overlay) CreateIdentityWithExternalId(t *testing.T, name, externalID, a
...
+	o.CreateLocalPkiCA(t, name)
+
+	caCert := filepath.Join(o.pkiRoot(), name, "certs", name+".cert")
+	out, err := o.execZiti("edge create ca %s %s --auth --ottca --autoca", name, caCert)
+	require.NoError(t, err, "create ca %s", name)
+	id := string(bytes.TrimSpace(out))
+
+	// --cacert/--cakey makes the CLI fetch the verificationToken and mint the CN=token cert itself
+	caKey := filepath.Join(o.pkiRoot(), name, "keys", name+".key")
+	_, err = o.execZiti("edge verify ca %s --cacert %s --cakey %s", name, caCert, caKey)
```
Comment (dovholuknf) [inline tests/integration/testutil/overlay.go]: i'd like to do this using the other path as well (possibly instead of) this path. meaning using the ziti cli to make a client cert and using `ziti edge verify ca <name> ( --cert <pemCertFile>` instead of cacert/key

File: tests/integration/testutil/overlay.go [hunk 3]
```diff
@@ -555,26 +567,135 @@ func (o *Overlay) CreateIdentityWithExternalId(t *testing.T, name, externalID, a
...
+	require.NoError(t, err, "pki create client %s for ca %s", commonName, caName)
+	cert, err := os.ReadFile(filepath.Join(o.pkiRoot(), caName, "certs", commonName+".cert"))
+	require.NoError(t, err, "read client cert for %s", commonName)
+	key, err := os.ReadFile(filepath.Join(o.pkiRoot(), caName, "keys", commonName+".key"))
+	require.NoError(t, err, "read client key for %s", commonName)
+	return string(cert), string(key)
+}
+
+// GetCaJwt fetches the CA enrollment JWT from the management API, authenticated
+// with the session the CLI cached at login. No ziti CLI verb exposes it.
```
Comment (dovholuknf) [inline tests/integration/testutil/overlay.go:678]: > No ziti CLI verb exposes it.

pleaes file an issue to expose it.

File: tests/integration/third_party_ca_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		caID := state.overlay.CreateThirdPartyCA(t, "test_tpca_neg")
+		caJwt := state.overlay.GetCaJwt(t, caID)
+		state.overlay.CreateLocalPkiCA(t, "test_tpca_unregistered")
+		cert, key := state.overlay.CreateClientCert(t, "test_tpca_unregistered", "test_tpca_unregistered_user1")
+
+		identityData := testutil.NewCaIdentityData("test_tpca_unregistered_user1", caJwt, cert, key)
+		addResp := state.zetClient.AddIdentity(t, identityData)
+		addResp.AssertFail(500, "key/cert are invalid")
+	})
+}
```
Comment (dovholuknf) [inline tests/integration/third_party_ca_test.go:148]: needs moar tests. adding the same identity twice, having no external id that matches - there seem like a lot of negative tests we could run (and maybe more positive tests)

**Review by dovholuknf — APPROVED** [review]
i'd like more tests (positive and negative) if we can get em but i'll approve as is
